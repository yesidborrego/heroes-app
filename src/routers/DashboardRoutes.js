// import React, { lazy, Suspense } from 'react'
import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'

import DcScreen from '../components/dc/DcScreen'
import HeroScreen from '../components/heroes/HeroScreen'
import MarvelScreen from '../components/marvel/MarvelScreen'
import Navbar from '../components/ui/Navbar'
import SearchScreen from '../components/search/SearchScreen'

/*
const DcScreen = lazy(() => import('../components/dc/DcScreen'));
const HeroScreen = lazy(() => import('../components/heroes/HeroScreen'));
const MarvelScreen = lazy(() => import('../components/marvel/MarvelScreen'));
const Navbar = lazy(() => import('../components/ui/Navbar'));
const SearchScreen = lazy(() => import('../components/search/SearchScreen'));
*/

export const DashboardRoutes = () => {
  return (
    <div>
      {/*<Suspense fallback={<div>Loading...</div>}>*/}
        <Navbar/>

      <div className="container mt-3">
        <Switch>
          <Route exact path="/marvel" component={MarvelScreen} />
          <Route exact path="/dc" component={DcScreen} />
          <Route exact path="/hero/:heroId" component={HeroScreen} />
          <Route exact path="/search" component={SearchScreen} />

          <Redirect to="/marvel" />
        </Switch>
      </div>
    {/*</Suspense>*/}
    </div>
  )
}
