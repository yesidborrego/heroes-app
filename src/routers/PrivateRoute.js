import React from 'react';
import PropTypes from 'prop-types';
import { Redirect, Route } from 'react-router-dom';

export const PrivateRoute = ({ isLogged, component: Component, ...rest }) => {
  const lastPatch = (rest.location.search.length === 0) ? rest.location.pathname : `${rest.location.pathname}${rest.location.search}`;
  localStorage.setItem('lastPatch', lastPatch);
  return (
    <Route {...rest} component={
      (props) => (
        (isLogged)
          ? ( <Component {...props} /> )
          : ( <Redirect to="/login" /> )
      )
    } />
  )
}

PrivateRoute.propTypes = {
  isLogged: PropTypes.bool.isRequired,
  component: PropTypes.func.isRequired
}