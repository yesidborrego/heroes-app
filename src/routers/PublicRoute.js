import React from 'react'
import { Redirect, Route } from 'react-router-dom'
import PropTypes from 'prop-types';

export const PublicRoute = ({ isLogged, component: Component, ...rest }) => {
  return (
    <Route
      {...rest}
      component={
        (props) => (
          (isLogged)
            ? <Redirect to="/" />
            : <Component {...props} />
        )
      }
    />
  )
}

PublicRoute.propTypes = {
  isLogged: PropTypes.bool.isRequired,
  component: PropTypes.func.isRequired
}