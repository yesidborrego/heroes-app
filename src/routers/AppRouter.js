import React, { useContext } from 'react';
import {
  BrowserRouter as Router,
  Switch
} from 'react-router-dom';
import { AuthContext } from '../auth/AuthContext';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {
  const {user} = useContext(AuthContext);

  return (
    <Router>
      <div>
        <Switch>
          {/*<Route exact path="/login" component={LoginScreen} />*/}
          <PublicRoute
            exact
            path="/login"
            isLogged={user.logged}
            component={LoginScreen}
          />
          {/*<Route path="/" component={DashboardRoutes} />*/}
          <PrivateRoute
            path="/"
            isLogged={user.logged}
            component={DashboardRoutes} />
        </Switch>
      </div>
    </Router>
  )
}
