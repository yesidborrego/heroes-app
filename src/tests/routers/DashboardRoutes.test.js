import '@testing-library/jest-dom';
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/AuthContext";
import { DashboardRoutes } from "../../routers/DashboardRoutes";

describe('Probar el componente <DashboardRoutes/>', () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: 'Isabella Borrego'
    }
  };

  test('Debe mostrarse correctamente el componente', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter>
          <DashboardRoutes/>
        </MemoryRouter>
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.text-info').text().trim()).toBe(contextValue.user.name);
  })

})
