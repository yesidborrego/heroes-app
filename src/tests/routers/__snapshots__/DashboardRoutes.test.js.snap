// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar el componente <DashboardRoutes/> Debe mostrarse correctamente el componente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search Hero
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ms-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Isabella Borrego
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>
  <div
    className="container mt-3"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <hr />
      <div
        className="row row-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card m-3 border-danger"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Spider Man"
                  className="card-img"
                  src="./assets/heroes/marvel-spider.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-text"
                  >
                    Spider Man
                  </h5>
                  <a
                    href="/hero/marvel-spider"
                    onClick={[Function]}
                  >
                    See more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card m-3 border-danger"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Captain America"
                  className="card-img"
                  src="./assets/heroes/marvel-captain.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-text"
                  >
                    Captain America
                  </h5>
                  <a
                    href="/hero/marvel-captain"
                    onClick={[Function]}
                  >
                    See more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card m-3 border-danger"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Iron Man"
                  className="card-img"
                  src="./assets/heroes/marvel-iron.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-text"
                  >
                    Iron Man
                  </h5>
                  <a
                    href="/hero/marvel-iron"
                    onClick={[Function]}
                  >
                    See more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card m-3 border-danger"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Thor"
                  className="card-img"
                  src="./assets/heroes/marvel-thor.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-text"
                  >
                    Thor
                  </h5>
                  <a
                    href="/hero/marvel-thor"
                    onClick={[Function]}
                  >
                    See more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card m-3 border-danger"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Hulk"
                  className="card-img"
                  src="./assets/heroes/marvel-hulk.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-text"
                  >
                    Hulk
                  </h5>
                  <a
                    href="/hero/marvel-hulk"
                    onClick={[Function]}
                  >
                    See more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card m-3 border-danger"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Wolverine"
                  className="card-img"
                  src="./assets/heroes/marvel-wolverine.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-text"
                  >
                    Wolverine
                  </h5>
                  <a
                    href="/hero/marvel-wolverine"
                    onClick={[Function]}
                  >
                    See more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card m-3 border-danger"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Daredevil"
                  className="card-img"
                  src="./assets/heroes/marvel-daredevil.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-text"
                  >
                    Daredevil
                  </h5>
                  <a
                    href="/hero/marvel-daredevil"
                    onClick={[Function]}
                  >
                    See more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card m-3 border-danger"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Hawkeye"
                  className="card-img"
                  src="./assets/heroes/marvel-hawkeye.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-text"
                  >
                    Hawkeye
                  </h5>
                  <a
                    href="/hero/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    See more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card m-3 border-danger"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Cyclops"
                  className="card-img"
                  src="./assets/heroes/marvel-cyclops.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-text"
                  >
                    Cyclops
                  </h5>
                  <a
                    href="/hero/marvel-cyclops"
                    onClick={[Function]}
                  >
                    See more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card m-3 border-danger"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Silver Surfer"
                  className="card-img"
                  src="./assets/heroes/marvel-silver.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-text"
                  >
                    Silver Surfer
                  </h5>
                  <a
                    href="/hero/marvel-silver"
                    onClick={[Function]}
                  >
                    See more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
